ENIGMAsystem/SHELL/Widget_Systems/Win32/dialogs.cpp

```diff
@@ -558,37 +558,51 @@ int get_color(int defcolor, bool advanced)
 
 string get_directory(string dname, string caption)
 {
-//NOTE: This uses the Windows Vista or later file chooser, which is different than the one used by GM8 and lower
-//because I could not find out which one it uses, since IFileDialog is used by both wxWidgets and QtFramework
-//and there doesn't appear to be a standard file picker for XP or lower in the Windows API except SHBrowseForFolder that is
-//used by Game Maker for get_directory_alt
-  IFileDialog* fileDialog;
-  CoCreateInstance(CLSID_FileOpenDialog, NULL, CLSCTX_INPROC_SERVER, IID_PPV_ARGS(&fileDialog));
+  //NOTE: This uses the Windows Vista or later file chooser, which is different than the one used by GM8 and lower
+  //because I could not find out which one it uses, since IFileDialog is used by both wxWidgets and QtFramework
+  //and there doesn't appear to be a standard file picker for XP or lower in the Windows API except SHBrowseForFolder that is
+  //used by Game Maker for get_directory_alt
+  IFileDialog *SelectDirectory;
```

Comment:
Please don't use CamelCase for variable names, those are reserved for types and sometimes functions. I don't know any style guide that recommends CamelCase variable names, so please don't do that.

Reply:
Your original code used camel case, but i can fix that for you as well.

Reply:
Ok, I'll change it.

Reply:
I meant don't use _upper_ CamelCase for variable names, lower camelCase is totally fine as is snake case in the context of variable names. Upper CamelCase is acceptable for types like VertexBuffer. snake_case is the only one generally ok for user functions or variables. Sometimes Upper CamelCase is acceptable for class member functions/fundies likes it and used it in `EnigmaPlugin.hpp` for emake.

Reply:
Resolved: 7f1f03547f166239450aa1b2f067fc1c59f37ecd

You fixed it.